std/math/exponential.d

```diff
@@ -112,44 +113,22 @@ if (isFloatingPoint!(F) && isIntegral!(G))
 
     enum uint bias = F.max_exp - 1;
 
-    static if (is(F == float))
-    {
-        float f = cast(float) v;
-        uint ival = () @trusted { return *cast(uint*) &f; }();
-        ulong ex = (ival >> 23) & 255;
-    }
-    else static if (is(F == double) || (is(T == real) && T.mant_dig == double.mant_dig))
+    ulong ex;
+    if (__ctfe)
     {
-        double d = cast(double) v;
-        ulong ival = () @trusted { return *cast(ulong*) &d; }();
-        ulong ex = (ival >> 52) & 2047;
-    }
-    else static if (is (F == real) && real.mant_dig == 64)
-    {
-        ulong ex = void;
-        if (__ctfe)
-        {
-            // in CTFE we cannot access the bit patterns and have therefore to
-            // fall back to the (slower) general case
-            // skipping subnormals by setting ex = bias
-            ex = fabs(v) == F.infinity ? 2 * bias + 1 :
-                (fabs(v) < F.min_normal ? bias : cast(ulong) (floor(log2(fabs(v))) + bias));
-        }
-        else
-        {
-            ulong[2] ival = () @trusted { return *cast(ulong[2]*) &v; }();
-            ex = ival[1] & 32767;
```

Comment:
`real.sizeof == 12` on at least some x86 systems. Reinterpeting as 2 longs and reading the 2nd one exceeds the allocated stack size and is invalid.